### PR TITLE
Rails 5 backports

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Moves some changes from the Rails 5 branch to master.

#### What should we test?

These changes are covered pretty well in existing specs, but here's a couple of things we could check:

- Quick sanity check that image uploads work
- Check connecting a stripe account works
- Basic enterprise registration works

#### Release notes

Backport various Rails 5 compatibility fixes.

Changelog Category: Technical changes

### CODE REVIEW

**Review by mkllnk — APPROVED** [review]
Thank you! :pray: 

This is so much more manageable. Our test coverage is not 100% and I searched for uses of `params` in general, just scanning through the code and guessing where there could be problems. I found a few uses of `Spree::Config.set(params[:preferences])`. Only one of four controllers was covered. I added another commit to your pull request. I hope that was okay. Please review.

I also saw `params[:file]` which seems to return a ActionDispatch::Http::UploadedFile object correctly in Rails 5. :heavy_check_mark: 

Accessing arrays seems fine, too. And I was worried about `ransack(params[:q])` but ransack seems to handle that well. :relieved:

**Comment by mkllnk** [comment]
Oh, by the way, I tested my commit on your pull request and on Rails 5. I did that first with your other branch and found that difficult. Then I remembered dual boot and tried that. I had to stop spring but after that it worked great for seeing the different behaviour between the systems. I don't like the complicated setup and Gemfile_next.lock is out of sync but otherwise it works great for me.

File: app/controllers/admin/enterprises_controller.rb [hunk 1]
```diff
@@ -64,12 +64,14 @@ def update
     end
 
     def register
-      if params[:sells] == 'unspecified'
+      register_params = params.permit(:sells)
+
+      if register_params[:sells] == 'unspecified'
```
[the remark this answers is not in the document]
Comment (Matt-Yorkley) [reply app/controllers/admin/enterprises_controller.rb:69]: On this line yes, but we also pass the params to an `#update` call in a line below, which will fail if it hasn't been permitted.

**Review by andrewpbrett — APPROVED** [review]
This looks like great work and less risky than the larger commit. I don't have much to add on top of Maikel's review other than I think it looks ready for some testing!

**Review by mkllnk — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley ,

Checked on the test cases you pointed out by creating a new enterprise, with images, new products, OC, shipping- and payment (Stripe-SCA) methods, from scratch. I think this covers them all: 

- Quick sanity check that image uploads work
- Check connecting a stripe account works
- Basic enterprise registration works

First order and checkout towards rails 5.0 was a hit :rocket: 

Ready to go!

**Pull request merged**